java/api/src/main/java/org/ray/api/id/BaseId.java

```diff
@@ -41,13 +41,14 @@ public ByteBuffer toByteBuffer() {
    */
   public boolean isNil() {
     if (isNilCache == null) {
-      isNilCache = true;
+      boolean localIsNil = true;
       for (int i = 0; i < size(); ++i) {
         if (id[i] != (byte) 0xff) {
-          isNilCache = false;
+          localIsNil = false;
           break;
         }
       }
+    isNilCache = localIsNil;
```

Comment:
oops. didn't notice there's a lint issue here. (why wasn't this caught by checkstyle?)